src/shogun/lib/sg_type_traits.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Saatvik Shah
+ */
+
+#include <type_traits>
+
+#ifndef SHOGUN_SG_TYPE_TRAITS_H
+#define SHOGUN_SG_TYPE_TRAITS_H
+/**
+ * A collection of useful type traits
+ */
+
+// Checks if any one of the types matches
+// Ref: https://stackoverflow.com/a/17032517/3656081
+template <typename T, typename... Rest>
+struct is_any_of : std::false_type
+{
+};
+
+template <typename T, typename First>
+struct is_any_of<T, First> : std::is_same<T, First>
+{
+};
+
+template <typename T, typename First, typename... Rest>
+struct is_any_of<T, First, Rest...>
+    : std::integral_constant<bool, std::is_same<T, First>::value ||
```

Comment:
btw I think I defined `is_sg_base` somewhere...we could move that into this file

Reply:
yep, ive moved it over